graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt

```diff
@@ -65,7 +65,7 @@ class DataFetcherInvoker internal constructor(
             return ReflectionUtils.invokeMethod(bridgedMethod, dgsComponent)
         }
 
-        if (kotlinFunction != null) {
+        if (dgsComponent.javaClass.getDeclaredAnnotation(Metadata::class.java) != null && kotlinFunction != null) {
```

Comment:
I think it'd be better to do this in the constructor to avoid having to check it every time. We can also use `KotlinDetector` from Spring. e.g.:

```kotlin
private val kotlinFunction: KFunction<*>? = if (KotlinDetector.isKotlinType(method.declaringClass)) bridgedMethod.kotlinFunction else null
```

Reply:
Addressed in a separate PR: https://github.com/Netflix/dgs-framework/pull/1697